react-catalog/my-app/src/pages/Catalog.tsx

```diff
@@ -0,0 +1,30 @@
+import { Product } from '../lib/data';
+import { useState, useEffect } from 'react';
+import { ProductCard } from '../components/ProductCard';
+import { readCatalog } from '../lib/read';
+
+export function Catalog() {
+  const [products, setProducts] = useState<Product[]>([]);
+  useEffect(() => {
+    const fetchProducts = async () => {
+      const data = await readCatalog();
+      setProducts(data);
+      console.log('products fetched');
+    };
+    try {
+      fetchProducts();
+    } catch (error) {
+      console.error(error);
+    }
+  }, [products]);
```

Comment:
We only want this `useEffect` to run on mount, so no need to have `products` as a dependency here. We would just have an empty dependency array. By including `products`, you're actually forcing the component to re-render more times than necessary.